lib/rules/array-bracket-spacing.js

```diff
@@ -130,19 +85,19 @@ module.exports = function(context) {
 
         if (isSameLine(first, second)) {
             if (openingBracketMustBeSpaced && !isSpaced(first, second)) {
-                reportRequiredBeginningSpace(node, first);
+                reportError(node, context.errors.spaceBefore, first);
```

Comment:
I don't think `context` is should be doing this. It's job is to provide information that isn't otherwise available in the file. This info exists in the file already.